agent/tcs/handler/handler.go

```diff
@@ -179,13 +181,18 @@ func anyMessageHandler(client wsclient.ClientServer) func(interface{}) {
 }
 
 // formatURL returns formatted url for tcs endpoint.
-func formatURL(endpoint string, cluster string, containerInstance string) string {
+func formatURL(endpoint string, cluster string, containerInstance string, taskEngine engine.TaskEngine) string {
 	tcsURL := endpoint
 	if !strings.HasSuffix(tcsURL, "/") {
 		tcsURL += "/"
 	}
 	query := url.Values{}
 	query.Set("cluster", cluster)
 	query.Set("containerInstance", containerInstance)
+	query.Set("agentVersion", version.Version)
+	query.Set("agentHash", version.GitHashString())
+	if dockerVersion, err := taskEngine.Version(); err == nil {
```

Comment:
If there is an error and we don't set dockerVersion, will the dockerVersion just be empty string in TACS or actually there will be no this attribute?

Reply:
no attribute. this is inspired from ACS -
https://github.com/aws/amazon-ecs-agent/blob/f95f731bce74e6276f9fd8bfbb87a62d3a790ce8/agent/acs/handler/acs_handler.go#L434-L436